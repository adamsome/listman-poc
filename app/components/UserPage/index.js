import React from 'react'

//import ListTiles from '../../containers/ListTiles'
import Tiles from '../Tiles'
import UserProfile from '../UserProfile'
import BodyColumn from '../Layout/BodyColumn'
import Container from '../Layout/Container'

import './UserPage.scss'

const UserPage = ({ user, lists, isLoading,
                    isAdding, addError, addList }) => (
  <BodyColumn>
    <UserProfile user={user} isLoading={isLoading} />
    <Container spaceAbove>
      <h2 className={"title is-2 " + ((isLoading)
        ? " user-page__lists-heading--loading"
        : ""
      )}>
        Lists
      </h2>
      {
        (addError)
          ? <article className="message is-danger">
              <div className="message-body">
                <strong>Error: </strong>{addError}
              </div>
            </article>
          : null
      }
      <Tiles 
        entities={lists}
        isLoading={isLoading}
        isAdding={isAdding}
        addError={addError}
        add={addList}
      />
    </Container>
  </BodyColumn>
)

export default UserPage
