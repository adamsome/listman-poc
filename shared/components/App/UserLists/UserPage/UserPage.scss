.user-page__lists-heading--loading {
  color: #e6e6e6;
}
