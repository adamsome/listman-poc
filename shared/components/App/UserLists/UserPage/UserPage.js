import React from 'react'

import TileList from '../../../TileList'
import UserProfile from '../../../UserProfile'
import BodyColumn from '../../../Layout/BodyColumn'
import Container from '../../../Layout/Container'

import './UserPage.scss'

const UserPage = ({ user, lists, isLoading,
                    isAdding, addError, addList }) => (
  <BodyColumn>
    <UserProfile user={user} isLoading={isLoading} />
    <Container spaceAbove>
      <h2 className={"title is-2 " + ((isLoading)
        ? " user-page__lists-heading--loading"
        : ""
      )}>
        Lists
      </h2>
      {
        (addError)
          ? <article className="message is-danger">
              <div className="message-body">
                <strong>Error: </strong>{addError}
              </div>
            </article>
          : null
      }
      <TileList
        entities={lists}
        isLoading={isLoading}
        isAdding={isAdding}
        addError={addError}
        add={addList}
      />
    </Container>
  </BodyColumn>
)

export default UserPage
