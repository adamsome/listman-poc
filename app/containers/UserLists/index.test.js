import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import { UserLists } from './index'
import UserPage from '../../components/UserPage'

it('passes isLoading false when user defined', () => {
  const wrapper = shallow(
    <UserLists
      user={{
        username: "test_username",
        description: "description of user",
        avatar: "http://path/to/avatar",
      }}
      lists={[
        { title: "list title" }
      ]}
    />
  )
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('passes isLoading true when user undefined', () => {
  const wrapper = shallow(
    <UserLists />
  )
  const child = wrapper.find(UserPage).first()
  expect(child.props().isLoading).toBe(true)
})

it('fetches on mount', () => {
  const mockFetch = jest.fn()
  const wrapper = mount(
    <UserLists
      fetchLists={mockFetch}
      params={{userID: "test_username"}}
      user={{
        username: "test_username",
        description: "description of user",
        avatar: "http://path/to/avatar",
      }}
      lists={[
        { title: "list title" }
      ]}
    />
  )
  expect(mockFetch.mock.calls.length).toBe(1)
})
