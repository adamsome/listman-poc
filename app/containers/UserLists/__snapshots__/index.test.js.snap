exports[`test renders <NotFound> on error 1`] = `
<NotFound
  error="User not found" />
`;

exports[`test renders correctly 1`] = `
<UserPage
  addList={[Function]}
  isLoading={false}
  lists={
    Array [
      Object {
        "title": "list title",
      },
    ]
  }
  user={
    Object {
      "avatar": "http://path/to/avatar",
      "description": "description of user",
      "username": "test_username",
    }
  } />
`;
